src/connection.js

```diff
@@ -1245,6 +1286,7 @@ Connection.prototype.STATE = {
     name: 'SentClientRequest',
     exit: function() {
       this.clearRequestTimer();
+      this.tokenStreamParser.resume();
```

Comment:
> This boils down to Request classes decides if the Request is paused state. And Connection class decides what that actually means. Sounds like a reasonable separation to me.

Yes exactly. Whether the request is in the paused state or not is determined by the application. This is consistent from the API point of view. Whether the token parser transform and all the other internal transforms are in the paused state, is internal to the driver and not exposed to the API.